src/lib/resolvers/channel.ts

```diff
@@ -1,11 +1,12 @@
-import type { ChannelTypes } from '@sapphire/discord.js-utilities';
+import { ChannelMentionRegex, ChannelTypes } from '@sapphire/discord.js-utilities';
 import { container } from '@sapphire/pieces';
-import type { Message, Snowflake } from 'discord.js';
+import type { Message } from 'discord.js';
 import { Identifiers } from '../errors/Identifiers';
 import { err, ok, Result } from '../parsers/Result';
 
 export function resolveChannel(parameter: string, message: Message): Result<ChannelTypes, Identifiers.ArgumentChannelError> {
-	const channel = (message.guild ? message.guild.channels : container.client.channels).cache.get(parameter as Snowflake);
+	const channelId = ChannelMentionRegex.exec(parameter);
+	const channel = (message.guild ? message.guild.channels : container.client.channels).cache.get(channelId?.[1] ?? parameter);
```

Comment:
```suggestion
	const channelId = (ChannelMentionRegex.exec(parameter)?.[1] ?? parameter) as Snowflake;
	const channel = (message.guild ? message.guild.channels : container.client.channels).cache.get(channelId);
```

Reply:
Why the cast tho? `get` takes a string, not a Snowflake

Reply:
`Snowflake` is type of `string`.

Reply:
Because we might break it again and change the type to `${bigint}` in the future.